### PR TITLE
Moving Breakdown Key Bits to use Gf2

### PR DESCRIPTION
**Pull request by benjaminsavage**
Migrating the breakdown key over to use Gf2. It can remain in Gf2 all the way through the Sort, most of attribution (where it is unused) until aggregation. It's only at the very end of the attribution stage that it must be converted over to an arithmetic sharing so that we can multiply by the credit.

### CODE REVIEW

File: src/protocol/attribution/aggregate_credit.rs [hunk 1]
```diff
@@ -194,6 +196,10 @@ where
                                 let step = BitOpStep::from(check_idx);
                                 let c = c2.narrow(&step);
                                 let record_id = RecordId::from(i);
+                                /// 
+                                /// TODO: modulus convert "check" to an element in the large prime field here!
```
Comment (benjaminsavage) [inline src/protocol/attribution/aggregate_credit.rs:200]: @martinthomson - I'm stuck here. Need your help figuring out how to perform the conversion at this point.

**Comment by benjaminsavage** [comment]
New implementation:

dhat: Total:     3,045,987,503 bytes in 21,715,462 blocks
dhat: At t-gmax: 95,969,876 bytes in 366,873 blocks
dhat: At t-end:  19,767,307 bytes in 146,694 blocks


Main branch before this commit:

dhat: Total:     3,079,907,867 bytes in 22,197,980 blocks
dhat: At t-gmax: 96,799,456 bytes in 371,200 blocks
dhat: At t-end:  19,947,259 bytes in 149,774 blocks

**Comment by benjaminsavage** [comment]
Now the results for n=1000

New Implementation:

dhat: Total:     30,848,618,479 bytes in 218,734,067 blocks
dhat: At t-gmax: 481,161,818 bytes in 1,551,603 blocks
dhat: At t-end:  20,047,415 bytes in 149,527 blocks


Main branch before this commit:

dhat: Total:     30,881,338,978 bytes in 223,534,011 blocks
dhat: At t-gmax: 482,564,634 bytes in 1,548,316 blocks
dhat: At t-end:  20,226,703 bytes in 152,603 blocks

**Review by martinthomson — APPROVED** [review]
There are some pieces that might be dead code now.  It might be worth looking for those.

File: src/protocol/attribution/input.rs [hunk 2]
```diff
@@ -130,7 +130,7 @@ impl<F: Field, T: LinearSecretSharing<F>> MCCreditCappingInputRow<F, T> {
     pub fn new(
         is_trigger_report: T,
         helper_bit: T,
-        breakdown_key: Vec<T>,
+        breakdown_key: Vec<Replicated<Gf2>>,
```
[the remark this answers is not in the document]
Comment (martinthomson) [reply src/protocol/attribution/input.rs:133]: I just looked into this and it's not clear that a change here would be a win.

**Pull request merged**